test/tests/selftest/pbs_requirements_decorator.py

```diff
@@ -89,3 +89,19 @@ def test_skip_no_server_on_mom(self):
         due to no_mom_on_server flag
         """
         self.server.expect(SERVER, {'server_state': 'Active'})
+
+    @requirements(num_servers=1, num_comms=1, no_comm_on_mom=True)
```

Comment:
'no_comm_on_mom' is True by default, this can be changed to False. Although this is not testing the change in json fields, it is an extended test for requirements decorator. I think test case -TestJSONReport.test_json_fields (pbs_json_test_report.py) can be extended to verify requirements field; which would validate the data populated in PTLJsonData.